CMakeLists.txt

```diff
@@ -140,26 +140,23 @@ if(BUILD_DOC)
 endif()
 
 find_package(PkgConfig)
-pkg_check_modules(SQLITE3 REQUIRED sqlite3)
-pkg_check_modules(BOTAN2 REQUIRED botan-2)
-pkg_check_modules(LIBUSB REQUIRED libusb-1.0)
-pkg_check_modules(GNUTLS REQUIRED gnutls)
+pkg_check_modules(sqlite REQUIRED IMPORTED_TARGET sqlite3)
+pkg_check_modules(botan-2 REQUIRED IMPORTED_TARGET botan-2)
+pkg_check_modules(libusb-1.0 REQUIRED IMPORTED_TARGET libusb-1.0)
+pkg_check_modules(gnutls REQUIRED IMPORTED_TARGET gnutls)
```

Comment:
This seems like a lot of magic. Needs more explanation

Reply:
Hm weird... I already answered on the but both are some where lost :-( ... ok in short again:

It is possible to import the third party library as target via pkg_check_modules() with the `IMPORTED_TARGET`.

The big advantage about the imported target is to use the transitive usage requirements from the target. This means, you should only add the third party library as link library, for example the botan libary:
```
target_link_libraries(foo ...
   ...
   PkgConfig::botan-2
   ...
 )
```
You didn't need to add the include path via `target_include_directories()` or the compile definitions via `target_compile_definitions()`. The include path and compile definitons are set via the imported target `PkgConfig::botan-2` on `target_link_libraries()`. As additional, botan has dependencies to BZip2 and ZLib and these libraries will be added also to the linker at link time.

See also in the cmake documentation [imported targets](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#imported-targets) and [transitive usage requirements](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#transitive-usage-requirements)

Or I also recommended to show the talks from C++Now 2017: Daniel Pfeifer “Effective CMake" and CppCon 2017: Mathieu Ropert “Using Modern CMake Patterns to Enforce a Good Modular Design”. lambdafu also mentioned this here #83